ADA/databricks_rstudio_sql_warehouse.qmd

```diff
@@ -1,105 +1,226 @@
+:::
 
-    ii) On the 'User DSN' tab click the 'Add...' button.
+-   Make a note of the 'Databricks access token' it has given you
 
-    iii) In the 'Create New Data Source' window, select 'Simba Spark ODBC Driver' and click 'Finish'.
+::: callout-warning
+It is very important that you immediately copy the access token that you are given, as you will not be able to see it through Databricks again. If you lose this access token before pasting it into RStudio then you must generate a new access token to replace it.
+:::
 
-    iv) In the 'Simba Spark ODBC Driver DSN Setup' window,
+------------------------------------------------------------------------
 
-        a.  Enter a 'Data Source Name' and 'Description'. Choose a short and sensible data source name and note it down as this is what you will use to connect to Databricks through RStudio. *As you can set up more than one cluster on Databricks, use the description to make clear which cluster this connection is for. The description shown below describes that this connection is using an 8 core cluster on Databricks Runtime Environment 13.*
-        b.  Set the remaning options to the settings below.
-            -   Enter the 'Server Hostname' for your cluster in the 'Host(s):' field (you noted this down in step 2).
-            -   In the Port section, remove the default number and use the Port number you noted in step 2. 
-            -   Set the Authentication Mechanism to 'User Name and Password'.
-            -   Enter the word 'token' into the 'User Name:' field, then enter your 'Databricks access token' in the 'Password:' field.
-            -   Change the Thrift Transport option to HTTP.
-            -   Click the 'HTTP Options...' button and enter the 'HTTP Path' of your Databricks cluster, then click 'Okay'.
-            -   Click the 'SSL Options...' button and tick the 'Enable SSL' box, then click the 'OK' button.
-            -   ![](../images/odbc-driver-settings.png)
-        c.  Click the 'Test' button to verify the connection has worked. You should see the following message. *If you get an error here, repeat steps 5.e.i -- 5.e.ix again and ensure all the values are correct.*
+#### Cluster ID
 
-        ![](../images/databricks-test-connection.png)
+------------------------------------------------------------------------
 
-        d.  Click the 'OK' button to exit the 'Test Results' window, then the 'OK' button in the 'Simba Spark ODBC Driver DSN Setup' window.
+The Cluster ID is the cluster that you'll be using to run your code.
 
-5.  Connect through RStudio. Watch the below video and view the [ADA_RStudio_connect GitHub repo](https://github.com/dfe-analytical-services/ADA_RStudio_connect) for methods on connecting to Databricks and querying data from RStudio. 
+To obtain this, follow these steps:
 
+-   click 'Compute' in the Databricks menu, then click on your cluster's name
```

Comment:
For new users I think there's a step here to create their compute cluster. I can't remember what the process was but I remember it being fairly easy(?)

Reply:
Agree there should be a step to include this. It's 'Compute' > 'Create with DfE Personal Compute'.

Reply:
Aren't all SQL warehouses automatically created with a shared cluster attached? I'm sure I heard that somewhere

Reply:
This isn't my understanding.

You might be thinking of personal clusters which they've setup to use a shared cluster filtering service to get around some permission issues. 

My understand (which is limited because I don't use SQL warehouses at the moment because they lack functionality I use) is that a SQL warehouse is a separate thing to a cluster.